src/document/DocumentAnnotator.ts

```diff
@@ -0,0 +1,82 @@
+// @flow
+import BaseAnnotator, { Options } from '../common/BaseAnnotator';
+import BaseManager from '../common/BaseManager';
+import RegionManager from '../region/RegionManager';
+import { ANNOTATOR_EVENT, CLASS_ANNOTATIONS_LOADED } from '../constants';
+
+export default class DocumentAnnotator extends BaseAnnotator {
+    annotatedEl?: HTMLElement;
+
+    annotations: Record<string, []> = {};
+
+    managers: Map<string, BaseManager[]> = new Map();
+
+    constructor(options: Options) {
+        super(options);
+
+        this.hydrate(options);
+    }
+
+    getPages(): HTMLElement[] {
+        // TODO: Inject page/container elements from Preview SDK rather than DOM?
+        return this.annotatedEl ? Array.from(this.annotatedEl.querySelectorAll('.page')) : [];
+    }
+
+    hydrate({ file }: Options): void {
+        this.api
+            .fetchAnnotations(file.file_version.id)
+            .then(() => {
+                // TODO: Normalize response, set in store, and render
+            })
+            .catch(error => {
+                this.emit(ANNOTATOR_EVENT.error, error);
+            });
+    }
+
+    init(scale: number): void {
+        super.init(scale);
+
+        if (!this.rootEl) {
+            return;
+        }
+
+        this.annotatedEl = this.rootEl.querySelector('.bp-doc') as HTMLElement;
+        this.annotatedEl.classList.add(CLASS_ANNOTATIONS_LOADED);
+
+        this.render();
+    }
+
+    render(): void {
+        this.getPages()
+            .filter(({ dataset }) => dataset.loaded && dataset.pageNumber)
+            .forEach(pageEl => this.renderPage(pageEl));
```

Comment:
could this just be `.forEach(this.renderPage)`?

Reply:
I would have to make `renderPage` a bound property rather than a method, which doesn't seem necessary.